### ISSUE TITLE
Exceeding maximum property value size should raise an error and fail early

### ISSUE THREAD

**Issue by ojlyytinen** [projecthydra/active_fedora#916]
It seems that there is a maximum size a property value can have and it seems to be 4095 characters (or bytes?). If you try to set values larger than that things start to break down. For example

```ruby
class Foo < ActiveFedora::Base
  property :title, predicate: ::RDF::DC.title
end

foo = Foo.new
foo.title = ['a'*4096]
foo.save
# Everything's fine up to here, you can even get the oversize title back intact.
# But if you manipulate the title further, old values will get stuck in Fedora.
foo.title = ['test']
foo.save
# at this point foo.title will be ['a'*4096,'test'] instead of just ['test']
```

If the property has multiple: false, then this breaks the whole record because ActiveFedora won't load it at all with multiple values.

The maximum size for properties seems quite reasonable. In cases where an application really needs massive properties the application logic can split it into parts or something. But at the moment when you exceed the limit everything appears to be fine but then weird unexpected things start to happen later. Raising an error when setting an oversize value would fix this.

**Comment by dchandekstark** [comment]
We should investigate what's happening in Fedora.

**Comment by ojlyytinen** [comment]
Yes definitely. It seems that this goes deeper than just ActiveFedora. When you have one of these oversize triples in the record, I'm unable to remove any triples using the same predicate even using the Fedora admin pages. Doesn't give any error messages, just won't remove the triples.

**Comment by mjgiarlo** [comment]
Ping @escowles @cbeer @awoods

**Comment by escowles** [comment]
Some background: there's a configurable threshold (defaults to 4096), and values smaller than that are stored with the metadata and values that large or larger are stored with the binaries.  So a small thumbnail could be stored with the metadata, and a large literal value could be stored with the files.

There's a Fedora issue related to this, where some operations don't work with literals that are larger than the binary storage threshold: https://jira.duraspace.org/browse/FCREPO-1561

There is a PR to partially address this, but it hasn't been merged: https://github.com/fcrepo4/fcrepo4/pull/911
